### PR TITLE
Smart Views columns should stretch to fill the container width

### PR DESCRIPTION
**Pull request by macrotim**
Before the fix, the List View displays empty whitespace to the right of the right-most column. 

With the fix...
Screen Shot 2019-03-28 at 2 11 31 PM

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @macrotim has signed the CLA. Thanks for the pull request!

File: src/features/list-view/ListView.js [hunk 1]
```diff
@@ -32,6 +32,12 @@ type CellRendererArgs = {|
     style: Object,
 |};
 
+type ComputeColumnWidthArgs = {|
```
Comment (ChenCodes) [inline src/features/list-view/ListView.js]: Could we move this and CellRendererArgs to a flowType.js file?

Comment (macrotim) [reply src/features/list-view/ListView.js]: Glad you pointed it out! I'll do that.

Comment (macrotim) [reply src/features/list-view/ListView.js]: Done.

File: src/features/list-view/ListView.js [hunk 2]
```diff
@@ -25,12 +27,7 @@ type Props = {
     width: number,
 };
 
-type CellRendererArgs = {|
-    columnIndex: number,
-    key: string,
-    rowIndex: number,
-    style: Object,
-|};
+export const DEFAULT_COLUMN_WIDTH = 300;
```
Comment (Visylvius) [inline src/features/list-view/ListView.js]: can this be moved to constants?

File: src/features/list-view/ListView.js [hunk 3]
```diff
@@ -67,8 +64,51 @@ class ListView extends React.PureComponent<Props> {
...
+        }
+
+        const firstColumnWidth = getClientDefinedColumnWidth(0);
+        if (columnIndex === 0) {
+            return firstColumnWidth;
+        }
+
+        // Determine if columns fill width of
+        // MultiGrid or stretching is needed.
+        let canFillWidth = false;
```
Comment (Visylvius) [inline src/features/list-view/ListView.js]: nitpick: can we move var declarations to the top of the function?

Comment (macrotim) [reply src/features/list-view/ListView.js]: Can you help me understand why you prefer that? 

Moving it to the top means any reader must jump their eyeballs up and down the function to make sense of it.

File: src/features/list-view/ListView.js (hunk 2 above)
Comment (macrotim) [reply src/features/list-view/ListView.js]: Done.

File: src/features/list-view/ListView.js (hunk 3 above)
Comment (Visylvius) [reply src/features/list-view/ListView.js]: Thats true, i've always followed the pattern of putting all var declarations at the top of the function. That way they're in one place and you always know where to look.

**Pull request merged**